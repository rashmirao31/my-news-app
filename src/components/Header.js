import React from 'react'
import Moment from 'react-moment';
import 'moment-timezone';
import CountryList from './CountryList.js';

class Header extends React.Component {

  render() {

    return (
      <div data-testid='header-menu'>
        <div className='editionDetails'>
          <div>
              <CountryList 
                updateSelectedCountry={this.props.updateSelectedCountry}
                updateSearchText={this.props.updateSearchText}
              />
          </div>
          <div>
          {window.defaultAppSettings.headerTimeLabel}
          <Moment unix tz={window.defaultAppSettings.timezoneKey} data-testid='time-label'>
                {this.props.updatedTime}
          </Moment>
          </div>
        </div>
      </div>
    );
  };
  
}

export default Header