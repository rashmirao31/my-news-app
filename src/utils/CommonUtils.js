var CommonUtils = {

	getConfigList: function() {
		let newsApiUrl = "https://newsapi.org/v2/";
		return {
			"newsApiUrl": newsApiUrl,
			"newsApiTopHeadlinesUrl": newsApiUrl + "top-headlines",
			"newsApiEverythingUrl": newsApiUrl + "everything",
			"apiKey": "687cf8fdc1d7435eb10214b70c78788d",
			"countryKey": "gb",
			"countryName": "United Kingdom",
			"category": "general",
			"timezoneKey": "Europe/London",
			"countryOptions": [
			  { key: 'gb', value: 'gb', flag: 'gb', text: 'United Kingdom' },
			  { key: 'us', value: 'us', flag: 'us', text: 'USA' },
			  { key: 'in', value: 'in', flag: 'in', text: 'India' }
			],
			"countryMenuTitle": "Country:",
			"countryMenuPlaceholder": "Select Country",
			"headerTimeLabel": "Updated at: ",
			"pageTitleSuffixLabel": "News",
			"searchInputPlaceholder": "Search...",
			"searchButtonText": "Search",
			"topStoriesText": "Top Stories",
			"searchResultHeaderText": "Search Results For",
			"loadingText": "Please wait...",
			"noDataText": "No results found",
			"readMoreText": "Read more",
			"errorScenarioText": "Something went wrong. Please try again after sometime."
		};
	},


};

export default CommonUtils;