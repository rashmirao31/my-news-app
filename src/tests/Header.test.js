import React from 'react';
import ReactDOM from 'react-dom';
import CommonUtils from './../utils/CommonUtils';
import Header from './../components/Header.js';

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

beforeEach(() => {
  window.defaultAppSettings = CommonUtils.getConfigList();
  return;
});

afterEach(cleanup);

it('renders without crashing', () => {
 	const div = document.createElement('div');
	ReactDOM.render(<Header />, div);
});

it('matches snapshot', () => {
 	const tree = renderer.create(
    	<Header/>
	).toJSON();
 	expect(tree).toMatchSnapshot();
});

it('renders with the updated time header label', () => {
 	const { getByText } = render(
    	<Header />
	);
	const updatedTimeLabel = window.defaultAppSettings.headerTimeLabel.trim();
	expect(getByText(updatedTimeLabel)).toBeVisible();
});

it('renders with the country menu title', () => {
 	const { getByText } = render(
    	<Header />
	);
	const countryMenuTitle = window.defaultAppSettings.countryMenuTitle;
	expect(getByText(countryMenuTitle)).toBeInTheDocument();
});

it('renders with the country list', () => {
 	const { getByTestId } = render(
    	<Header />
	);
	const element = getByTestId('country-list');
	expect(element).toBeInTheDocument();
});

it('renders with the updated time', () => {
 	const updatedTime = Math.round((new Date()).getTime() / 1000);
 	const { getByTestId } = render(
    	<Header updatedTime={updatedTime}/>
	);
	const element = getByTestId('time-label');
	expect(element).not.toHaveTextContent('Invalid date');
});