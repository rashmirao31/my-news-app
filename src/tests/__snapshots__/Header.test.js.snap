// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="header-menu"
>
  <div
    className="editionDetails"
  >
    <div>
      <div>
        Country:
        <div
          aria-expanded={false}
          className="ui search selection dropdown"
          data-testid="country-list"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="combobox"
        >
          <input
            aria-autocomplete="list"
            autoComplete="off"
            className="search"
            onChange={[Function]}
            style={
              Object {
                "width": undefined,
              }
            }
            tabIndex={0}
            type="text"
            value=""
          />
          <div
            aria-atomic={true}
            aria-live="polite"
            className="text"
            role="alert"
          >
            United Kingdom
          </div>
          <i
            aria-hidden="true"
            className="dropdown icon"
            onClick={[Function]}
          />
          <div
            className="menu transition"
            role="listbox"
          >
            <div
              aria-checked={true}
              aria-selected={true}
              className="active selected item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <i
                className="gb flag"
              />
              <span
                className="text"
              >
                United Kingdom
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <i
                className="us flag"
              />
              <span
                className="text"
              >
                USA
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <i
                className="in flag"
              />
              <span
                className="text"
              >
                India
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      Updated at: 
      <time
        data-testid="time-label"
        dateTime={null}
      >
        Invalid date
      </time>
    </div>
  </div>
</div>
`;
